File: app/javascript/utils/company.js
export const formatResourceCost = ({man, mun, fuel}) => {
  let cost = ""
  if (man && man !== 0) {
    cost += `${man}MP `
  }
  if (mun && mun !== 0) {
    cost += `${mun}MU `
  }
  if (fuel && fuel !== 0) {
    cost += `${fuel}FU`
  }

  if (cost.length === 0) {
    cost += "Free"
  }
  return cost
}

File: spec/factories/company.rb
FactoryBot.define do
  factory :company do
    association :faction
    association :doctrine
    association :player
    association :ruleset
    name { "Company name" }
    sequence :uuid do |n| "uuid #{n}" end
    type { "ActiveCompany" }
    vps_earned { 0 }
    vps_current { 0 }
    pop { 0 }
    man { 7000 }
    mun { 1600 }
    fuel { 1400 }
  end
end

